pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -154,12 +148,8 @@ def restore_model(self) -> None:
 
     def restore_model_weights(self, checkpoint_path: Optional[_PATH]) -> None:
         """Restore only the model weights."""
-        checkpoint = self._loaded_checkpoint
```

Comment:
- when is checkpoint_path `None` ? 
- i haven't traced through the paths fully here. is there a concern we're now loading a different checkpoint dict somewhere?

Reply:
its never None, `restore_model_weights` is only called with `trainer._ckpt_path` if `trainer._ckpt_path` is not None. should we remove the None check for clarity?

Reply:
in that case i believe we should. @awaelchli what do you think?